### PR TITLE
Fix 2 bug

### PR DESCRIPTION
**Pull request by Moanrisy**
*Image got cropped in here

https://i.ibb.co/Zgt2Bhp/bug1.png
https://i.ibb.co/JyqvY3X/fix1.png
https://i.ibb.co/ncsKrkR/bug2.png
https://i.ibb.co/YLsFnXm/fix2.png

1. Bug: Current prayer time show 0:0 after Ishaa time
SS of bug 1 

After fixing bug 1 

2. Bug: Dummy value used as current prayer time when current time is midnight (00h:00m:00s until Fajr time)
SS of bug 2 

After fixing bug 2

### CODE REVIEW

File: src/time.rs [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
 
 use crate::{Date, DateTime};
 
 pub fn now() -> DateTime {
     Local::now().naive_local()
 }
 
+pub fn one_sec_before_midnight() -> DateTime {
+    let midnight = NaiveTime::from_hms_opt(23, 59, 59);
+    Local::now().date_naive().and_time(midnight.unwrap())
```
Comment (azzamsa) [inline src/time.rs:11]: Avoid using unwrap here, use `?`.

File: src/time.rs [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+}
+
+pub fn one_sec_after_midnight() -> DateTime {
+    let midnight = NaiveTime::from_hms_opt(0, 0, 01);
+    Local::now().date_naive().and_time(midnight.unwrap())
```
Comment (azzamsa) [inline src/time.rs:16]: Same thing.

File: src/salah/times.rs [hunk 3]
```diff
@@ -312,7 +319,7 @@ impl PrayerTimes {
     fn current_time(&self, time: DateTime) -> Prayer {
         // dummy value. it will replaced below
         // just to avoid using `Option` or `Err`
-        let mut current_prayer = Prayer::Dohr;
+        let mut current_prayer = Prayer::Ishaa;
```
Comment (azzamsa) [inline src/salah/times.rs:322]: Would you like to make sure this change doesn't cause regressions?

Or, make sure that the failed test is valid. I use `https://www.jadwalsholat.org/`
See https://github.com/azzamsa/islam/blob/7f19f2274c8da578bb92d9dd090eec5f9a3d301c/src/salah/times.rs#L429, https://github.com/azzamsa/islam/blob/7f19f2274c8da578bb92d9dd090eec5f9a3d301c/src/salah/times.rs#L377

**Comment by azzamsa** [comment]
@Moanrisy it is better for you to create separate PR for each bug. I see nothing wrong with your first commit.

**Comment by azzamsa** [comment]
@Moanrisy I will make a changes in https://github.com/azzamsa/islam/tree/patch, then see if this solves your issue.

**Comment by azzamsa** [comment]
@Moanrisy still interested? Your fix is critical for other users. I am very concerned to release a hot fix.

**Comment by Moanrisy** [comment]
I'm still interested, but I'm little busy right now. Maybe Wednesday I can work on it.

File: src/time.rs (hunk 1 above)
Comment (Moanrisy) [reply src/time.rs:11]: Got it.  Already resubmit PR with Option using ? instead of unwrap.

File: src/salah/times.rs (hunk 3 above)
Comment (Moanrisy) [reply src/salah/times.rs:322]: Fixed test current_prayer_is_dohr()
by replacing expected_time method with expected_time_with_date.
Other prayer test is using the later one, I wonder why only the Dohr using expected_time.

let current_prayer_time = expected_time_with_date(date, 11, 52, 0)?;

**Pull request closed without merging**